### PR TITLE
Add example manifests

### PR DESCRIPTION
**Pull request by mattkelly**
#### What does this pull request accomplish?

 Add example manifests for core Cerebral components. The goal of this is to enable a trivial "Getting Started" flow.

 There are a few tradeoffs with how I did this, which I'm happy to discuss further.

 - Use `examples` dir over `deploy` as it seems more aligned with the community at this time
 - Combine some manifests, e.g. all CRDs and other prereqs, into one file
   - This is to enable a `kubectl apply -f https://raw.github.com/...` workflow, since it's not possible to apply a directory this way

 #### What issue(s) does this fix?
* None

 #### Additional Considerations

 ### Testing

 In progress.

 ### Dependencies
* [x] #93 should probably be merged first.

### CODE REVIEW

**Comment by mattkelly** [comment]
I intend to remove the `deploy` directory entirely and add ASG and ASP examples to the `examples` dir after initial feedback that this is an ok approach (or not) for now.

**Comment by mattkelly** [comment]
Also need to add a top-level `README.md` to `examples` explaining:

- File structure
- Manifests there (currently) reference `latest` and thus should not be blindly used for production

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,5 @@
 SHELL=/bin/bash
+:wq
```
Comment (normanjoyner) [inline Makefile]: 🤘

File: examples/engines/containership/10-deployment-cerebral-containership.yaml [hunk 2]
```diff
@@ -8,19 +8,22 @@ metadata:
     containership.io/app: cerebral
     containership.io/managed: "true"
     app.kubernetes.io/name: cerebral
+    app.kubernetes.io/part-of: cerebral
```
Comment (normanjoyner) [inline examples/engines/containership/10-deployment-cerebral-containership.yaml]: 👍

File: Makefile [hunk 3]
```diff
@@ -74,9 +75,9 @@ release: ## Build release image for controller (must be on semver tag)
 
 ### Commands for local development
 .PHONY: deploy
-deploy: ## Deploy the controller
-	kubectl apply -f deploy/development/cerebral.yaml
+deploy: ## Deploy Cerebral (without any prereqs)
+	kubectl apply -f examples/engines/containership/10-deployment-cerebral-containership.yaml
```
Comment (ashleyschuett) [inline Makefile]: why `10-`?

File: Makefile (hunk 1 above)
Comment (mattkelly) [reply Makefile]: wow

File: examples/engines/aws/10-deployment-cerebral-aws.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    spec:
+      serviceAccountName: cerebral
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      containers:
+      - name: cerebral
+        image: containership/cerebral:latest
```
Comment (ashleyschuett) [inline examples/engines/aws/10-deployment-cerebral-aws.yaml]: should we make a note in the read me that the manifests will run the latest image and is not versioned?

Comment (mattkelly) [reply examples/engines/aws/10-deployment-cerebral-aws.yaml]: Yup, this is a WIP - see my [above comment](https://github.com/containership/cerebral/pull/94#issuecomment-462176666). I'm really just putting this up early to solicit feedback on the general approach. This was probably unclear.

File: Makefile (hunk 3 above)
Comment (mattkelly) [reply Makefile]: All of the manifests are prefixed with an integer to ensure a sane apply order. This is similar to the approach taken by other projects such as `velero`. I forgot to clarify this in the description, whoops!

An alternative approach would be to split logical units into separate directories, but that makes the apply-from-url approach more cumbersome (see original description). I'm happy to reconsider this approach though.

File: examples/metrics_backends/influxdb/00-metrics-backend-influxdb.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+apiVersion: cerebral.containership.io/v1alpha1
```
Comment (mattkelly) [inline examples/metrics_backends/influxdb/00-metrics-backend-influxdb.yaml]: Add `---` as first line

File: examples/engines/digitalocean/20-autoscaling-engine-digitalocean.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+apiVersion: cerebral.containership.io/v1alpha1
```
Comment (mattkelly) [inline examples/engines/digitalocean/20-autoscaling-engine-digitalocean.yaml]: Add `---` as first line

File: examples/engines/aws/00-secret-cerebral-aws.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
```
Comment (mattkelly) [inline examples/engines/aws/00-secret-cerebral-aws.yaml]: Add `---` as first line

File: examples/common/00-prereqs.yaml [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+          required:
+            - type
+          properties:
+            type:
+              type: string
+            configuration:
+              type: object
+
+
+---
```
Comment (mattkelly) [inline examples/common/00-prereqs.yaml]: Convention of two spaces between manifests stolen from `velero`. It seemed kinda nice 🤷‍♂️

**Comment by ashleyschuett** [comment]
I like the overall structure. I'm kinda of thinking it'd be nice to not duplicate the docs folder structure within the examples folder and put them together but keep going back and forth on this.

**Comment by mattkelly** [comment]
@ashleyschuett I agree that the duplicated structure is an unfortunate artifact of this approach. I also don't love that we have to (IMO) remove the example manifests from the docs in order to avoid duplication and link to the relevant example manifests instead. It's just a bit more indirection. I almost wish you could embed linked yaml in the markdown. But for now it seems like a reasonable compromise to me 🤷‍♂️

File: examples/engines/containership/10-deployment-cerebral-containership.yaml (hunk 2 above)
Comment (mattkelly) [reply examples/engines/containership/10-deployment-cerebral-containership.yaml]: This is from https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/ btw

**Comment by mattkelly** [comment]
This is updated and ready for final review(s).

GitHub doesn't have a tree view, so here's one for easier context:

```
➜ tree examples
examples
├── README.md
├── autoscaling_groups
│   ├── README.md
│   ├── containership-prometheus.yaml
│   ├── kops-kubernetes.yaml
│   └── kubernetes-cpu-allocation.yaml
├── autoscaling_policies
│   ├── README.md
│   ├── influxdb-cpu-utilization.yaml
│   ├── prometheus-cpu-utilization.yaml
│   └── prometheus-custom-cpu-utilization.yaml
├── common
│   ├── 00-prereqs.yaml
│   └── README.md
├── engines
│   ├── aws
│   │   ├── 00-secret-cerebral-aws.yaml
│   │   ├── 10-deployment-cerebral-aws.yaml
│   │   ├── 20-autoscaling-engine-aws.yaml
│   │   └── README.md
│   ├── containership
│   │   ├── 00-secret-cerebral-containership.yaml
│   │   ├── 10-deployment-cerebral-containership.yaml
│   │   ├── 20-autoscaling-engine-containership.yaml
│   │   └── README.md
│   └── digitalocean
│       ├── 00-secret-cerebral-digitalocean.yaml
│       ├── 10-deployment-cerebral-digitalocean.yaml
│       ├── 20-autoscaling-engine-digitalocean.yaml
│       └── README.md
└── metrics_backends
    ├── influxdb
    │   ├── 00-metrics-backend-influxdb.yaml
    │   └── README.md
    ├── kubernetes
    │   ├── 00-metrics-backend-kubernetes.yaml
    │   └── README.md
    └── prometheus
        ├── 00-metrics-backend-prometheus.yaml
        └── README.md
```

Pretty brutal in terms of number of files tbh. I also forgot that there are already great ASP examples in the [metrics backend docs](https://github.com/containership/cerebral/tree/master/docs/metrics_backends). My theory for now is the more docs the merrier and we can cut later as we figure out what makes sense / get user feedback. I tried to cross-reference docs as best as I could (and verify that those links work).

File: examples/common/00-prereqs.yaml [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+      properties:
+        spec:
+          required:
+            - type
+          properties:
+            type:
+              type: string
+            configuration:
+              type: object
+
```
Comment (ashleyschuett) [inline examples/common/00-prereqs.yaml]: one more space please

File: examples/engines/aws/00-secret-cerebral-aws.yaml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cerebral-aws-engine
+  namespace: kube-system
+data:
```
Comment (ashleyschuett) [inline examples/engines/aws/00-secret-cerebral-aws.yaml]: I think this should be stringData so the user doesn't need to  base64 decode their keys

File: examples/engines/containership/00-secret-cerebral-containership.yaml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cerebral-containership-engine
+  namespace: containership-core
+data:
```
Comment (ashleyschuett) [inline examples/engines/containership/00-secret-cerebral-containership.yaml]: stringData

File: examples/engines/digitalocean/00-secret-cerebral-digitalocean.yaml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cerebral-digitalocean-engine
+  namespace: kube-system
+data:
```
Comment (ashleyschuett) [inline examples/engines/digitalocean/00-secret-cerebral-digitalocean.yaml]: stringData

File: examples/metrics_backends/kubernetes/README.md [hunk 13]
```diff
@@ -0,0 +1,8 @@
+# File Structure
+
+## 00-metrics-backend-kubernetes.yaml
+
+This file contains a MetricsBackend CustomResource for registering the Kubernetes backend with Cerebral.
+No additional configuration is required.
+
```
Comment (ashleyschuett) [inline examples/metrics_backends/kubernetes/README.md]: maybe make note that the example service URL that is in the example is assuming Prometheus is running in the default namespace?

File: examples/common/00-prereqs.yaml (hunk 9 above)
Comment (mattkelly) [reply examples/common/00-prereqs.yaml]: Good eyes!

File: examples/engines/aws/00-secret-cerebral-aws.yaml (hunk 10 above)
Comment (mattkelly) [reply examples/engines/aws/00-secret-cerebral-aws.yaml]: I understand what this means now. Interesting - I had no idea you could do this. Makes sense to me I guess 🤷‍♂️

File: examples/metrics_backends/kubernetes/README.md (hunk 13 above)
Comment (mattkelly) [reply examples/metrics_backends/kubernetes/README.md]: I think you mixed up the files while reviewing. This is the Kubernetes backend. The Prometheus backend already has a note about the namespace :)

**Comment by mattkelly** [comment]
@ashleyschuett @normanjoyner updated.

File: examples/metrics_backends/kubernetes/README.md (hunk 13 above)
Comment (ashleyschuett) [reply examples/metrics_backends/kubernetes/README.md]: you  are correct 👍

**Pull request merged**